### ISSUE TITLE
Multiple outlines, shadows and gradients for FlxText

### ISSUE THREAD

**Issue by JoeCreates** [HaxeFlixel/flixel#1165]
Presently, it is very difficult to make a `FlxText` look distinct. I wish to improve `FlxText` to make it possible to have multiple outlines, multiple shadows, gradients, and perhaps other effects. What I would like would be a breaking change, so before I go ahead and try it, I'd like some feedback.

I would like to change the following:-
- Remove `borderStyle` and its related properties. (Since when did a shadow count as a border?)
- Make a general `FlxTextEffect` to represent the various filters.
- Make a `FlxTextOutline`, `FlxTextGradient`, and `FlxTextShadow`.
- Add an array of `FlxTextEffect`s to `FlxText`.
- Add shorthand methods, `addOutline()`, `addShadow()`, and `addGradient()`?

Note that effects would be applied cumulatively. This would make text significantly more customizable. Instead of just having a single border or outline, you could have outlined text with a shadow, or shadowed text with an outline around the shadow etc..
##

**Comment by adrianulima** [comment]
Seeing this issue now. What if this new class `FlxEffectText` extends the newly released `FlxEffectSprite` and use the `IFlxEffect`s? The missing effects could added and would work for every `FlxEffectSprite` as well, and I'm sure you guys could also help to improve existing effects.

**Comment by Gama11** [comment]
Using a separate `FlxText` class for this doesn't sound good.

**Comment by adrianulima** [comment]
Right, not a new class so. Passing current `FlxText` as target to `FlxEffectSPrite` would work?
